File: app/src/main/java/com/mlieou/yaara/model/File.java
package com.mlieou.yaara.model;

import java.util.List;

/**
 * Created by mlieou on 1/3/18.
 */

public class File {

    private int index;
    private String path;
    private long length;
    private long completedLength;
    private boolean selected;
    private List<Uri> uris;

    public int getIndex() {
        return index;
    }

    public String getPath() {
        return path;
    }

    public long getLength() {
        return length;
    }

    public long getCompletedLength() {
        return completedLength;
    }

    public boolean isSelected() {
        return selected;
    }

    public List<Uri> getUris() {
        return uris;
    }

    public class Uri {
        public String uri;
        public String status;
    }
}

File: app/src/main/java/com/mlieou/yaara/model/File.kt
package com.mlieou.yaara.model

/**
 * Created by mlieou on 5/19/18.
 */

class File(val index : Int,
           val path : String,
           val length : Long,
           val completedLength : Long,
           val selected : Boolean,
           val uris : List<Uri>)

class Uri(val uri: String, val status : String)